rust/ql/lib/codeql/rust/dataflow/internal/ModelsAsData.qll

```diff
@@ -0,0 +1,98 @@
+/**
+ * Defines extensible predicates for contributing library models from data extensions.
+ */
+
+private import rust
+private import codeql.rust.dataflow.FlowSummary
+
+/**
+ * Holds if in a call to the function with canonical path `path`, defined in the
+ * crate `crate`, the value referred to by `output` is a flow source of the given
+ * `kind`.
+ *
+ * `output = "ReturnValue"` simply means the result of the call itself.
+ *
+ * The following kinds are supported:
+ *
+ * - `remote`: a general remote flow source.
```

Comment:
The values discussed in https://github.com/github/codeql/blob/main/docs/codeql/reusables/threat-model-description.rst should be available, in particular `remote`, `local` and the sub-types of `local` (e.g. we're already using `commandargs` and `environment` in `Concepts.qll`).